### PR TITLE
Implement instant refineries.

### PR DESCRIPTION
**Pull request by abcdefg30**
AKA RA2 refinery logic.

Credit for this goes to @GraionDilach: https://github.com/AttacqueSuperior/Engine/commit/8cf23c4605e4532bfd7f2856ea366cadc5336658

### CODE REVIEW

**Comment by pchote** [comment]
This will end up being removed again when somebody finally takes on the job of fixing our resource system, but I guess it is okay as a stop-gap.

File: OpenRA.Mods.Common/Traits/Buildings/Refinery.cs [hunk 1]
```diff
@@ -37,6 +37,9 @@ public class RefineryInfo : IAcceptResourcesInfo, Requires<WithSpriteBodyInfo>
 		[Desc("In how many steps to perform the dragging?")]
 		public readonly int DragLength = 0;
 
+		[Desc("Store resources in silos.")]
```
Comment (rob-v) [inline OpenRA.Mods.Common/Traits/Buildings/Refinery.cs]: is probably clear, but it might say sth like 'else give cash'.

**Review by rob-v — APPROVED** [review]
👍

File: OpenRA.Mods.Common/Traits/Buildings/Refinery.cs (hunk 1 above)
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/Buildings/Refinery.cs]: Updated. Hope it's more clear now.

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/774779df4dae3365e4bb8d0d9d8f71d63ef9f0e6...f6a11fbaeb72dbf61c55d785e7bbb116a635acf1

**Pull request merged**